builtin/providers/aws/resource_aws_kms_alias.go

```diff
@@ -46,7 +61,16 @@ func resourceAwsKmsAlias() *schema.Resource {
 
 func resourceAwsKmsAliasCreate(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).kmsconn
-	name := d.Get("name").(string)
+
+	var name string
+	if v, ok := d.GetOk("name"); ok {
+		name = v.(string)
+	} else if v, ok := d.GetOk("name_prefix"); ok {
+		name = resource.PrefixedUniqueId(v.(string))
+	} else {
+		name = resource.UniqueId()
+	}
+
```

Comment:
Do I need to make any change here?  The name was in the old implementation so I just keep it.  I saw similar implementation with security_group.  Please let me know if I need to fix this for the merge or it's something we can deal with with another pull request.